NuRadioReco/modules/io/NuRadioRecoio.py

```diff
@@ -296,6 +296,11 @@ def get_detector(self):
                     # Detector is a generic detector, so we have to consider default
                     # station/channel and event-specific changes
                     self.__detectors[self._current_file_id] = NuRadioReco.detector.generic_detector.GenericDetector.__new__(NuRadioReco.detector.generic_detector.GenericDetector)
+                    # the use of default_station and default_channel is deprecated. Allow to set it for now, to ensure backward compatibility
+                    if 'default_station' not in detector_dict:
+                        detector_dict['default_station'] = None
+                    if 'default_channel' not in detector_dict:
+                        detector_dict['default_channel'] = None
                     self.__detectors[self._current_file_id].__init__(source='dictionary', json_filename='', dictionary=detector_dict, default_station=detector_dict['default_station'], default_channel=detector_dict['default_channel'])
```

Comment:
I don't think you need these lines. You can just not pass any default channels and stations to the generic detector __init__ (and should, because these are no longer supposed to be passed).

Reply:
Never mind, I get what you are doing now. Looks good to me.